odo/convert.py

```diff
@@ -213,14 +222,29 @@ def _():
             yield convert(pd.DataFrame, [], **kwargs)
     else:
         df = convert(pd.DataFrame, first, **kwargs)
+        df1, n1 = mkindex(df, 0)
 
         def _():
-            yield df
+            n = n1
+            yield df1
             for i in rest:
-                yield convert(pd.DataFrame, i, **kwargs)
+                df = convert(pd.DataFrame, i, **kwargs)
+                df, n = mkindex(df, n)
+                yield df
     return chunks(pd.DataFrame)(_)
 
 
+def _ignore_index(df, start):
+    return df, start
+
+
+def _add_index(df, start):
+    stop = start + len(df)
+    idx = pd.RangeIndex(start=start, stop=stop)
```

Comment:
range index is added in pandas 17, unfortunatly some people still need to support pandas 16, could we conditionally define this based on the prescense of `RangeIndex`?